[Gan Jing Wen] iP

gan-jw opened the pull request.

Commit:
Add Gradle support

gan-jw committed:
Implemented a skeletal version of DeD that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

gan-jw committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

gan-jw committed:
Add the ability to mark tasks as done.

gan-jw committed:
Fixed bug where only the first word of user input is stored.

gan-jw committed:
Add support for tracking three types of tasks: Todos, Deadlines and Events.

gan-jw committed:
Fixed bug: todo display

gan-jw committed:
Fixed typos

gan-jw committed:
Used the input/output redirection technique to semi-automate the testing of Duke.

gan-jw committed:
Teach Duke to deal with errors such as incorrect inputs entered by the user.

gan-jw committed:
Added support for deleting tasks from the list.

gan-jw committed:
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up.

gan-jw committed:
Teach Duke how to understand dates

gan-jw committed:
Merge branch 'branch-Level-8'

gan-jw committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

gan-jw committed:
fix bug from merge conflict

gan-jw committed:
Refactor the code to extract out closely related code as classes

gan-jw committed:
Organize the classes into suitable java packages.

gan-jw committed:
Improve OOP

gan-jw committed:
Organize the classes into more java packages

Commit:
build.gradle: Update version to 8.29

erinmayg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+package main.java.duke;
+
+
+import main.java.duke.command.*;
```
I think it is better to explicitly state the imported classes that you're going to use instead of using '*' 😄

erinmayg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+
```
I think it complies with the coding standard more if you add "// Fallthrough" at the end of each cases with no "break" line.

erinmayg commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+        new Duke("data/tasks.txt").run();
+    }
+
+
```
LGTM 😄 I like the clean coding style.

erinmayg commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+            while (scanner.hasNextLine()) {
+                String[] currLine = scanner.nextLine().split(" \\| ");
+                String typeOfTask = currLine[0];
+                switch (typeOfTask) {
```
I like how you use switch statements 😄 It's easier to read compared to 'ifs and elses'.

erinmayg commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+                    }
+                    list.add(event);
+                    break;
+                default:
```
I think it's better to omit the default if it contains no statements, it makes the code simpler.

erinmayg commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public String listAllTasks() {
+        String res = "";
```
Perhaps you could use a StringBuilder instead? I think it makes it run faster. It's optional though 😅

kkangs0226 replied:
I agree, using StringBuilder with append() method would improve the runtime of your code!!:)

erinmayg commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        return taskList.size();
+    }
+
+
```
LGTM 😄 Other than a few small (optional) changes.

erinmayg commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    public void printMessage(String message) {
+        System.out.println(String.format(message));
+    }
+
```
I like the clean coding style 😄 LGTM!

erinmayg commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,30 @@
+
+public class AddCommand extends Command {
+    private Task task;
+    public static final String COMMAND_WORD_TODO = "todo";
```
I think attributes should be private (and if used outside the class, try adding a get method)... Noticed it, in other Command classes as well. I like how you're using it to improve code readability 😄  Other than that, LGTM!

erinmayg commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+    public static final String COMMAND_WORD = "exit";
+    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "bb cya again!";
+
+    public ExitCommand() {}
```
You can omit this 😄 I think Java has automatically built an empty constructor method for every object... Noticed it in other types of commands as well.

erinmayg commented on src/main/java/duke/command/OwoCommand.java:
```diff
@@ -0,0 +1,17 @@
+import main.java.duke.TaskList;
+import main.java.duke.Ui;
+
+public class OwoCommand extends Command {
```
It's cute, I love it! 🥰 LGTM!

erinmayg commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import main.java.duke.tasks.Task;
```
Perhaps you could add a blank space to separate import statements from different packages? I think it complies better with the CS2103 coding standard 😄 Noticed it in other classes as well.

erinmayg commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+        try {
+            this.date = LocalDate.parse(date);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Time format should be yyyy/mm/dd or yyyy-mm-dd.");
```
I like how you specified the correct format! It lessens the confusion for new (possible) users! 😄

erinmayg commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+package main.java.duke;
+
+import main.java.duke.command.Command;
+
```
I noticed you haven't added Javadoc to any of the classes and methods 😅 Perhaps you could add it in the future?

erinmayg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+
+    public static Command parse(String userInput) throws DukeException {
+
+        String splitCommand[] = userInput.split(" ", 2);
```
I think it complies better with the Java coding standard to put the square brackets '[ ]' next to the data type 😅  Noticed it in other places as well.

erinmayg reviewed (commented):
Just a few changes needed to be made, after that LGTM 😄

Perpetual09 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,32 @@
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
+    public static final String MESSAGE_DELETE_ACKNOWLEDGEMENT =
+            "Nooo you can't take away what you've already given me...\n"
+                    + "Okay fine. It's in my stomach tho... ASDFGUUVHHH!!\n"
+                    + "The following has been removed: ";
```
Maybe can use StringBuilder to replace +

Perpetual09 commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,18 @@
+    public String saveToString() {
+        return String.format("%c | %s", type, super.saveToString());
+    }
+}
```
If I remember correctly, we need to do at least half of Javadoc for all these files. Maybe you can do that later

gan-jw committed:
add changes according to peer feedback:
-explicitly state the imported classes
-add "// Fallthrough" at the end of each cases with no "break" line
-omit the default if it contains no statements
-add a blank space to separate import statements from different packages
-omit redundant constructors in command classes

gan-jw committed:
Add JUnit tests to test the behavior of the code

gan-jw committed:
Add JavaDoc comments to the code

gan-jw committed:
Add find command to give users a way to find a task by searching for a keyword

kkangs0226 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,128 @@
+     * @param delete If the action is to delete the task in file.
+     * @throws DukeException  If unable to edit file.
+     */
+    public void editTaskList(String newTask, int taskNum, boolean delete) throws DukeException {
```
I think you could use a clearer name for boolean so that we know it is a boolean value such as isDelete or shouldDelete:)

kkangs0226 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+ * Represents the current the task list.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
```
It would be better to name the variable "tasks" in this case as we would want to differentiate between single and multi-valued variable!!

kkangs0226 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,42 @@
+
+    }
+
+    public String saveToString() {
```
I like the way you used %s %c for your string representations, ive never thought of doing this before!! It looks great:)

kkangs0226 reviewed (commented):
On the whole, your project LGTM:) Just very minor changes to be made, keep up the good work!

gan-jw committed:
Add the Gradle wrapper files to the project

gan-jw committed:
Fix import statements and packaging

gan-jw committed:
Set up gradle so that it can build and run Duke, and run unit tests using gradle

gan-jw committed:
Merge branch 'branch-a-gradle'

gan-jw committed:
Use checkStyle to detect coding style violations

gan-jw committed:
Merge branch 'branch-a-checkstyle'

gan-jw committed:
Add a GUI to Duke using the JavaFX technology

gan-jw committed:
Merge branch 'branch-Level-10'

gan-jw committed:
Improve coding style and generate jar file with gradle

gan-jw committed:
Use assert feature to document important assumptions that should hold at various points in the code

gan-jw committed:
Examine the code and refactor to improve the code quality where necessary

gan-jw committed:
Provide a way to leverage statistics about the items managed by the App

gan-jw committed:
Merge pull request #1 from gan-jw/branch-a-assertions

Use assert feature to document important assumptions

gan-jw committed:
Merge pull request #2 from gan-jw/branch-a-codequality

Examine the code and refactor to improve the code quality

gan-jw committed:
Merge branch 'master' into branch-c-statistics

gan-jw committed:
Merge pull request #3 from gan-jw/branch-c-statistics

Provide a way to leverage statistics about the items managed by the App

gan-jw committed:
Fix bug: incorrect arguments

gan-jw committed:
Improve code quality and refactoring of Parser class further

gan-jw committed:
Adjust ui picture displayed

gan-jw committed:
Add help command

gan-jw committed:
Add a User Guide

gan-jw committed:
Set theme jekyll-theme-merlot

gan-jw committed:
Improve the GUI to make it more polished

gan-jw committed:
Fix checkstyle

gan-jw committed:
Improve GUI and fix typos in readme

gan-jw committed:
Update README.md

gan-jw committed:
Merge branch 'master' of https://github.com/gan-jw/ip

gan-jw committed:
Merge branch 'master' of https://github.com/gan-jw/ip
